src/node/scan-dirs.ts

```
@@ -11,64 +11,109 @@ import pm from 'picomatch'
+    }
+
+    // Track bracket depth for all bracket types. Angle brackets are
+    // ambiguous (generics vs comparison operators). We use a heuristic:
+    // `<` is a generic opener when the last non-whitespace character is an
+    // identifier char, `)`, `>`, or `=` (for arrow fn generics like
+    // `= <T>(...)`). A comparison like `1 < 2` has a digit before
+    // whitespace, which matches `\w` — but importantly, a digit followed
+    // by ` < ` is always comparison. We exclude digits explicitly.
+    if (ch === '<') {
+      let k = i - 1
+      while (k >= 0 && /\s/.test(rest[k])) k--
```

ESLint

[error] Move this regular expression to module scope to avoid re-compilation on every call.

(e18e/prefer-static-regex)